airbyte-ci/connectors/cdk/cdk/java_cdk/tasks.py

```diff
@@ -0,0 +1,109 @@
+"""CI Workflow tasks for Airbyte Java CDK"""
+
+import os
+
+import pygit2
+from aircmd.actions.environments import with_gradle
+from aircmd.models.base import PipelineContext
+from aircmd.models.utils import load_settings, make_pass_decorator
+from dagger import Client, Container
+from prefect import task
+from prefect.artifacts import create_link_artifact
+
+from .settings import JavaCDKSettings
+
+pass_pipeline_context = make_pass_decorator(PipelineContext, ensure=True)
+pass_global_settings = make_pass_decorator(JavaCDKSettings, ensure=True)
+
+
+@task
+async def build_java_cdk_task(client: Client, settings: JavaCDKSettings, ctx: PipelineContext, scan: bool = False) -> Container:
```

Comment:
I notice a lot of repeatable code like:
* Declaration of a gradle command
* Running this gradle command
* Optionally add `--scan`
* rsync the cache

Can this be factored under a common GradleTask class, kinda like what I did here (not with `aircmd` though):
https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/pipelines/gradle.py#L16

Reply:
Yes, i'll file a ticket for this 👍

Reply:
@alafanechere and @cpdeethree - I just added some time on the calendar for Monday. I temporarily removed that code from `airbyte-ci/connectors/cdk` in this commit - and will plan to add back once we know the right path forward. 

(One option was to _not_ have a CDK-specific build package, but instead incorporate this as a command or set of tasks under the "connectors" flow.)

Bookmarking that code here for easy reference:

- https://github.com/airbytehq/airbyte/pull/28687/commits/241420757dbbc57d3e4ab815d43cdb9c76972190

Reply:
What I'm working on as of now is trying to get `connector-mysql` or `connector-mssql` to build in the CI engine. I think this command, for instance, works fine locally but not yet when called from the CI flow:

```
./gradlew :airbyte-integrations:connectors:source-mysql:integrationTest
```

Reply:
Latest CI run with CLI command `airbyte-ci connectors --name=source-mysql test` fails at this step:

`./gradlew --no-daemon --scan --build-cache :airbyte-integrations:connectors:source-mysql:distTar -x airbyteDocker`

But that command works fine locally. Failure when run through CI CLI is:

<img width="1059" alt="image" src="https://github.com/airbytehq/airbyte/assets/18150651/8b9e1f0a-7fbf-4285-9155-a6995dd4c30b">

Reply:
After a bit further thought I'm not sure having the task as a class is the best with how prefect works and how we reason about this, but I am sure we can come up with some way to abstract this that prevents us from having to do all these things. Perhaps with_gradle is extended to handle this stuff

Reply:
https://github.com/airbytehq/airbyte-platform-internal/issues/8182